core/src/main/java/tech/tablesaw/io/jdbc/SqlResultSetReader.java

```diff
@@ -115,10 +114,13 @@ public static Table read(ResultSet resultSet) throws SQLException {
 long, Long:         -9223372036854775808    9223372036854775807 NUMBER(18)      999_999_999_999_999_999
 
 */
-                    // Start with SHORT (since ColumnType.BYTE isn't supported yet)
-                    // and find the smallest java integer type that fits
-                    if (p <= 4) {
-                        type = ShortColumnType.instance();
+                    if (p <= 0) {
+                	// Default to double if invalid precision
+                	type = ColumnType.DOUBLE;
```

Comment:
Since both NUMERIC and DECIMAL types are mapped to DOUBLE in SQL_TYPE_TO_TABLESAW_TYPE, it might be better to actually not change the type from what it pulls from the map.  That way if if you use SqlResultSetReader.mapJdbcTypeToColumnType you can have NUMERIC treated as whatever type you want (and by default it would be DOUBLE).

note: I'll still be happy if it is just hardcoded to DOUBLE.

Reply:
Thanks for the review! I've updated per your suggestion. Please take another look